app/utills.py

```diff
@@ -0,0 +1,26 @@
+import httpx
+import os
+from dotenv import load_dotenv
+
+
+load_dotenv()
+
+API_KEY = os.getenv("API_KEY")
+BASE_URL = "http://api.openweathermap.org/data/2.5/weather"
+
+
+async def fetch_temperature(city_name: str) -> float:
+    params = {
+        "q": city_name,
+        "appid": API_KEY,
+        "units": "metric"
+    }
+    async with httpx.AsyncClient() as client:
+        try:
+            response = await client.get(BASE_URL, params=params)
+            response.raise_for_status()
+            data = response.json()
+            return data["main"]["temp"]
+        except httpx.HTTPStatusError as e:
+            print(f"Error fetching temperature for {city_name}: {e.response.status_code} - {e.response.text}")
```

Comment:
You're currently printing the error message and then raising the exception. This could lead to the same error message being printed multiple times if the exception is not caught higher up in the call stack. Consider logging the error message instead of printing it. This way, you can control the output of the error messages and avoid duplicate prints. Also, in a production environment, you would typically want to avoid printing any potentially sensitive information to the console.